lib/excon.rb

```diff
@@ -109,6 +109,7 @@ def new(url, params = {})
       end
       params = {
         :host       => uri.host,
+        :hostname   => uri.hostname,
```

Comment:
Yeah, if we could check ruby version and conditionally do it I think that would be preferred at least for the near future.